diff --git a/modules/dino_roar/src/Controller/RoarController.php b/modules/dino_roar/src/Controller/RoarController.php
index d63b069..5d80bd9 100644
--- a/modules/dino_roar/src/Controller/RoarController.php
+++ b/modules/dino_roar/src/Controller/RoarController.php
@@ -44,6 +44,8 @@ public function roar($count)
         $this->loggerFactoryService->get('default')
             ->debug($roar);
 
-        return new Response($roar);
+        return [
+            '#title' => $roar
+        ];
     }
 }
