diff --git a/modules/dino_roar/src/Controller/RoarController.php b/modules/dino_roar/src/Controller/RoarController.php
index 2ad60c2..ff0720c 100644
--- a/modules/dino_roar/src/Controller/RoarController.php
+++ b/modules/dino_roar/src/Controller/RoarController.php
@@ -30,6 +30,8 @@ public function roar($count)
     {
         $roar = $this->roarGenerator->getRoar($count);
 
-        return new Response($roar);
+        return [
+            '#title' => $roar
+        ];
     }
 }
