diff --git a/modules/dino_roar/src/Controller/RoarController.php b/modules/dino_roar/src/Controller/RoarController.php
index e12f0f9..624bffb 100644
--- a/modules/dino_roar/src/Controller/RoarController.php
+++ b/modules/dino_roar/src/Controller/RoarController.php
@@ -37,10 +37,11 @@ public function roar($count)
     {
         $keyValueStore = $this->keyValue('dino');
 
-        $roar = $this->roarGenerator->getRoar($count);
+        //$roar = $this->roarGenerator->getRoar($count);
+        //$keyValueStore->set('roar_string', $roar);
+        $roar = $keyValueStore->get('roar_string');
         $this->loggerFactory->get('default')
             ->debug($roar);
-        $keyValueStore->set('roar_string', $roar);
 
         return new Response($roar);
     }
