diff --git a/modules/dino_roar/src/Controller/RoarController.php b/modules/dino_roar/src/Controller/RoarController.php
index d63b069..2a76caa 100644
--- a/modules/dino_roar/src/Controller/RoarController.php
+++ b/modules/dino_roar/src/Controller/RoarController.php
@@ -40,9 +40,12 @@ public static function create(ContainerInterface $container)
 
     public function roar($count)
     {
+        $keyValueStore = $this->keyValue('dino');
+
         $roar = $this->roarGenerator->getRoar($count);
         $this->loggerFactoryService->get('default')
             ->debug($roar);
+        $keyValueStore->set('roar_string', $roar);
 
         return new Response($roar);
     }
