diff --git a/modules/dino_roar/dino_roar.services.yml b/modules/dino_roar/dino_roar.services.yml
index eb1126a..5a1b0ef 100644
--- a/modules/dino_roar/dino_roar.services.yml
+++ b/modules/dino_roar/dino_roar.services.yml
@@ -3,3 +3,4 @@ services:
         class: Drupal\dino_roar\Jurassic\RoarGenerator
         arguments:
             - '@keyvalue'
+            - true
diff --git a/modules/dino_roar/src/Jurassic/RoarGenerator.php b/modules/dino_roar/src/Jurassic/RoarGenerator.php
index 75c77ce..4ff596a 100644
--- a/modules/dino_roar/src/Jurassic/RoarGenerator.php
+++ b/modules/dino_roar/src/Jurassic/RoarGenerator.php
@@ -7,10 +7,12 @@
 class RoarGenerator
 {
     private $keyValueFactory;
+    private $useCache;
 
-    public function __construct(KeyValueFactoryInterface $keyValueFactory)
+    public function __construct(KeyValueFactoryInterface $keyValueFactory, $useCache)
     {
         $this->keyValueFactory = $keyValueFactory;
+        $this->useCache = $useCache;
     }
 
     public function getRoar($length)
@@ -18,14 +20,16 @@ public function getRoar($length)
         $store = $this->keyValueFactory->get('dino');
         $key = 'roar_'.$length;
 
-        if ($store->has($key)) {
+        if ($this->useCache && $store->has($key)) {
             return $store->get($key);
         }
 
         sleep(2);
 
         $string = 'R'.str_repeat('O', $length).'AR!';
-        $store->set($key, $string);
+        if ($this->useCache) {
+            $store->set($key, $string);
+        }
 
         return $string;
     }
